src/main/kotlin/com/terraformation/backend/tracking/ObservationService.kt

```diff
@@ -210,4 +234,30 @@ class ObservationService(
       throw InvalidObservationEndDateException(startDate, endDate)
     }
   }
+
+  private fun elapsedWeeks(source: Instant, weeks: Long): Boolean =
+      source.isBefore(clock.instant().minus(weeks * 7, ChronoUnit.DAYS).plus(1, ChronoUnit.HOURS))
+
+  private fun earliestPlantingElapsedWeeks(plantingSiteId: PlantingSiteId, weeks: Long): Boolean =
+      terrawareServerConfig.observations.notifyOnFirstPlanting &&
+          !observationStore.hasObservations(plantingSiteId) &&
+          (plantingSiteStore.fetchOldestPlantingTime(plantingSiteId)?.let {
+            elapsedWeeks(it, weeks)
+          }
+              ?: false)
+
+  private fun fetchNonNotifiedSitesForThresholds(
+      completedTimeElapsedWeeks: Long,
+      firstPlantingElapsedWeeks: Long,
+      siteIds: List<PlantingSiteId>
+  ): Collection<PlantingSiteId> {
+    requirePermissions { manageNotifications() }
```

Comment:
This is harmless, of course, but I usually only put permission checks on public methods (unless the private methods have more restrictive permissions than their callers).